### PR TITLE
Remove experimental entities filter

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
|-----------------|-----
| Branch?         |2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes (upgrade guide has been updated)
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -241,6 +241,45 @@
...
+                    type: entities
+                    label: sylius.ui.channel
+                    form_options:
+                        class: "%sylius.model.channel.class%"
+                    options:
+                        field: product.channels.id
+```
+
+After
+Before:
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md]:
```suggestion
After:
```

**Review by NoResponseMate — DISMISSED** [review]

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -239,9 +239,6 @@
         <service id="sylius.order_item_names_setter" class="Sylius\Component\Core\Order\OrderItemNamesSetter" />
         <service id="Sylius\Component\Core\Order\OrderItemNamesSetterInterface" alias="sylius.order_item_names_setter" />
 
-        <service id="sylius.grid_filter.entities" class="Sylius\Component\Core\Grid\Filter\EntitiesFilter">
```
Comment (loic425) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services.xml:242]: @GSadee Should I also mention that removed service in UPGRADE.md?

File: UPGRADE-2.0.md [hunk 3]
```diff
@@ -325,6 +325,44 @@
...
+            # ...
+            filters:
+                channel:
+                    type: entity
+                    label: sylius.ui.channel
+                    form_options:
+                        class: "%sylius.model.channel.class%"
+                    options:
+                        fields: [product.channels.id]
+```
```
Comment (GSadee) [inline UPGRADE-2.0.md]: As only 2 lines (if I see correclty) have changed, maybe the diff would be more transparent, wdyt? 🤔 
```suggestion
```diff
sylius_grid:
    grids:
        # ...
        sylius_admin_catalog_promotion:
            # ...
            filters:
                channel:
-                   type: entities
+                   type: entity
                    label: sylius.ui.channel
                    form_options:
                        class: "%sylius.model.channel.class%"
                    options:
-                       field: product.channels.id
+                       fields: [product.channels.id]
```
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services.xml (hunk 2 above)
Comment (GSadee) [reply src/Sylius/Bundle/CoreBundle/Resources/config/services.xml:242]: Yes, add a note to the UPGRADE file, and ideally also a deprecation to 1.14 🙊

**Review by GSadee — APPROVED** [review]

**Pull request merged**